controller/src/main/java/io/pravega/controller/server/bucket/PeriodicWatermarking.java

```diff
@@ -387,14 +391,28 @@ boolean checkExistsInCache(Stream stream) {
         private final ConcurrentHashMap<String, Long> inactiveWriters;
         
         WatermarkClient(Stream stream, ClientConfig clientConfig) {
-            this(stream, SynchronizerClientFactory.withScope(stream.getScope(), clientConfig));
+            this(stream, SynchronizerClientFactory.withScope(stream.getScope(), clientConfig), true);
         }
         
         @VisibleForTesting
         WatermarkClient(Stream stream, SynchronizerClientFactory clientFactory) {
-            this.client = clientFactory.createRevisionedStreamClient(
-                    NameUtils.getMarkStreamForStream(stream.getStreamName()), 
-                    new WatermarkSerializer(), SynchronizerConfig.builder().build());
+            this(stream, clientFactory, false);
```

Comment:
closeClientFactory is set to false only for tests. Hence we could remove the condition at line 411 below.

Reply:
removed closeclientfactory